core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/AbstractHealthCheckedEndpointGroupBuilder.java

```diff
@@ -201,6 +209,156 @@ public AbstractHealthCheckedEndpointGroupBuilder allowEmptyEndpoints(boolean all
+     *                               .selectionTimeout(Duration.ofSeconds(10), Duration.ofSeconds(3))
+     *                               ...
+     *                               .build();
+     * assert endpointGroup.selectionTimeoutMillis() == 13000;
+     *
+     * // Wait for the initial endpoints.
+     * endpointGroup.whenReady().join();
+     * assert endpointGroup.selectionTimeoutMillis() == 6000;
+     * }</pre>
+     *
+     * @param initialSelectionTimeout the initial selection timeout to wait for the
+     *                                {@linkplain HealthCheckedEndpointGroup#whenReady() initial endpoints}.
+     *                                If unspecified, {@link Flags#defaultResponseTimeoutMillis()} is used by
+     *                                default.
+     * @param selectionTimeout the selection timeout to wait for an {@link Endpoint} after the initial
+     *                         endpoints are resolved.
+     *                         If unspecified, {@link Flags#defaultConnectTimeoutMillis()}} is used by
+     *                         default.
+     */
+    @UnstableApi
+    public AbstractHealthCheckedEndpointGroupBuilder selectionTimeout(Duration initialSelectionTimeout,
+                                                                      Duration selectionTimeout) {
+        requireNonNull(initialSelectionTimeout, "initialSelectionTimeout");
+        requireNonNull(selectionTimeout, "selectionTimeout");
+        return selectionTimeoutMillis(initialSelectionTimeout.toMillis(), selectionTimeout.toMillis());
+    }
+
+    /**
+     * Sets the timeout to wait until a successful {@link Endpoint} selection.
+     * {@link Duration#ZERO} disables the timeout.
+     * If unspecified, {@link Flags#defaultResponseTimeoutMillis()} is used to wait for the
+     * {@linkplain HealthCheckedEndpointGroup#whenReady() initial endpoints}, and
+     * {@link Flags#defaultConnectTimeoutMillis()}} is used after the initial endpoints are resolved by default.
+     *
+     * <p>Note that the specified {@code selectionTimeoutMillis} and the delegate's
+     * {@linkplain EndpointGroup#selectionTimeoutMillis() selectionTimeoutMillis} are added and set to
+     * this {@link HealthCheckedEndpointGroup}.
+     * For example:<pre>{@code
+     * DnsAddressEndpointGroup delegate = DnsAddressEndpointGroup.builder("armeria.dev")
+     *                                                           .selectionTimeoutMillis(3000)
+     *                                                           ...
+     *                                                           .build();
+     * HealthCheckedEndpointGroup endpointGroup =
+     *     HealthCheckedEndpointGroup.builder(delegate, "/health")
+     *                               .selectionTimeoutMillis(10000)
+     *                               ...
+     *                               .build();
+     * assert endpointGroup.selectionTimeoutMillis() == 13000;
+     * }</pre>
```

Comment:
nit; what do you think of adding an explanation that setting this will set both `selectionTimeoutMillis` and `initialSelectionTimeoutMillis`